public/pages/Transforms/containers/Transforms/EditTransform.test.tsx

```diff
@@ -0,0 +1,137 @@
+                        render={(props: RouteComponentProps) => (
+                          <EditTransform {...props} transformService={services.transformService} core={coreServicesMock} />
+                        )}
+                      />
+                      <Route path={ROUTES.TRANSFORMS} render={(props) => <div>Testing transform edit page</div>} />
+                      <Redirect from="/" to={ROUTES.EDIT_TRANSFORM} />
+                    </Switch>
+                  </ModalProvider>
+                )
+              }
+            </ServicesConsumer>
+          </ServicesContext.Provider>
+        </CoreServicesContext.Provider>
+      </Router>
+    ),
+  };
+}
+
+describe("<EditTransform /> spec", () => {
+  it("renders the component", async () => {
+    browserServicesMock.transformService.getTransform = jest.fn().mockResolvedValue({
+      ok: true,
+      response: testTransform,
+    });
+    const { container } = renderEditTransformWithRouter([`${ROUTES.EDIT_TRANSFORM}?id=${testTransform._id}`]);
+
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  it("sets breadcrumbs when mounting", async () => {
+    browserServicesMock.transformService.getTransform = jest.fn().mockResolvedValue({
+      ok: true,
+      response: testTransform,
+    });
+    renderEditTransformWithRouter([`${ROUTES.EDIT_TRANSFORM}?id=${testTransform._id}`]);
+
+    expect(coreServicesMock.chrome.setBreadcrumbs).toHaveBeenCalledTimes(1);
+    expect(coreServicesMock.chrome.setBreadcrumbs).toHaveBeenCalledWith([
+      BREADCRUMBS.INDEX_MANAGEMENT,
+      BREADCRUMBS.TRANSFORMS,
+      BREADCRUMBS.EDIT_TRANSFORM,
+      { text: testTransform._id },
+    ]);
+  });
+
+  it("adds error toaster when get transform has error", async () => {
+    browserServicesMock.transformService.getTransform = jest.fn().mockResolvedValue({ ok: false, error: "some error" });
+    const { getByText } = renderEditTransformWithRouter([`${ROUTES.EDIT_TRANSFORM}?id=${testTransform._id}`]);
+
+    await waitFor(() => {});
+
+    expect(coreServicesMock.notifications.toasts.addDanger).toHaveBeenCalledTimes(1);
+    expect(coreServicesMock.notifications.toasts.addDanger).toHaveBeenCalledWith("Could not load transform job test1: some error");
+
+    await waitFor(() => getByText("Testing transform edit page"));
+  });
+
+  it("adds error toaster when get transform throws error", async () => {
+    browserServicesMock.transformService.getTransform = jest.fn().mockRejectedValue(new Error("rejected error"));
+    const { getByText } = renderEditTransformWithRouter([`${ROUTES.EDIT_TRANSFORM}?id=${testTransform._id}`]);
+
+    await waitFor(() => {});
+
+    expect(coreServicesMock.notifications.toasts.addDanger).toHaveBeenCalledTimes(1);
+    expect(coreServicesMock.notifications.toasts.addDanger).toHaveBeenCalledWith("rejected error");
+    await waitFor(() => getByText("Testing transform edit page"));
+  });
+
+  it("can edit description", async () => {
```

Comment:
I see that you mentioned this test is incomplete. I think we can probably create a issue with the test cases that are not implemented but should be. Right here I think that we can add check to make sure that the fields such as name is read only.

Reply:
That was on the previous revision, I made sure the test was actually testing what it needed to, because previously it just wasn't checking for anything it needed to. Checking for read-only fields would be a good future improvement for sure.